engine/src/flutter/shell/platform/darwin/ios/framework/Source/UITextInput+TextRange.swift

```diff
@@ -0,0 +1,221 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import UIKit
+
+@objc(FlutterTextPosition)
+final class TextPosition: UITextPosition, NSCopying {
+  let offset: UInt
+  @objc let affinity: UITextStorageDirection
+
+  init(index: UInt, affinity: UITextStorageDirection = .forward) {
+    self.offset = index
+    self.affinity = affinity
+  }
+
+  override func isEqual(_ object: Any?) -> Bool {
+    guard let other = object as? TextPosition else {
+      return false
+    }
+    // Affinity is only visual.
```

Comment:
can we double check UITextInput's position and ensure different affinity indeed returns equal?